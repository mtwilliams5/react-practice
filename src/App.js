import React, { Component } from 'react';
import request from 'request-promise';
import logo from './logo.svg';
import './App.css';
import UpdateUserForm from './UpdateUser';

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      user: {}
    }
    this.getGitHubData = this.getGitHubData.bind(this);
  }

  getGitHubData(user) {
    request.get(`https://api.github.com/users/${user}`)
      .then(data =>
        this.setState(() => {
          return { user: JSON.parse(data) }
        })
      );
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>

        <UpdateUserForm submitAction={this.getGitHubData} /> { /* You can pass functions down for child components to invoke */ }

        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
