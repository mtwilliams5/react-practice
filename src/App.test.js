import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('a test block', () => {
  it('should be 42', () => {
    expect(40 + 2).toEqual(42);
  })

  it('should match snapshot', () => {
    expect(40 + 2).toMatchSnapshot();
  });
});