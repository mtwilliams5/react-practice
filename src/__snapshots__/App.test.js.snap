// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a test block should match snapshot 1`] = `42`;
